WordPress/Learn
# Add Lesson Plan Details and Actions blocks

See #274 

This ports the Lesson Plan details and actions into 2 block widgets, so that they can be moved in the sidebar by an admin via the widget editor. The structure of the code is copied from the [Workshop Details block](https://github.com/WordPress/Learn/tree/trunk/wp-content/plugins/wporg-learn/js/workshop-details).

There are minimal changes on the frontend; buttons styles have been reused for the Actions block and HTML semantics and accessibility have been improved in the Details block. See screenshots below.

Some validation has been added to the existing Workshop Details block widget to stop the Widgets editor crashing if it is added to the sidebar for an incompatible post type.

### Screenshots

| Widget Editor | Validation |
|-|-|
| ![Screen Shot 2022-09-20 at 5 54 00 PM](https://user-images.githubusercontent.com/1017872/191178141-48abcc91-eb6a-496c-af2c-ae145db9a1f0.jpg) | ![Screen Shot 2022-09-20 at 5 52 59 PM](https://user-images.githubusercontent.com/1017872/191178192-1596f13f-ef91-429f-9792-bc4a1125f216.jpg) |

| Desktop before | Desktop after |
|-|-|
| ![Screen Shot 2022-09-19 at 11 53 48 AM](https://user-images.githubusercontent.com/1017872/190933515-b5682a62-c639-44c3-85ed-a629057b81af.jpg) | ![Screen Shot 2022-09-19 at 11 54 02 AM](https://user-images.githubusercontent.com/1017872/190933523-c9cfd3b0-3552-496f-b995-2085c7b2fe32.jpg) |

| Mobile | Print |
|-|-|
| ![Screen Shot 2022-09-19 at 11 40 43 AM](https://user-images.githubusercontent.com/1017872/190933086-0c38475f-8ad6-471a-a68c-a40596b2dab9.jpg) | ![Screen Shot 2022-09-19 at 11 41 27 AM](https://user-images.githubusercontent.com/1017872/190933104-6beceead-94e7-4ec6-97e9-65b28a92f3b7.jpg) |

### How to test this PR
1. Visit http://localhost:8888/wp-admin/widgets.php and expand the Lesson Plan widget.
2. There should be 2 new blocks available; Lesson Plan Details and Lesson Plan Actions. Add them and test reordering the sidebar, updating and checking the frontend display, comparing to prod.
3. Test posts with different media; slides linked and slides downloadable.
4. Test with a small screen < 768px; the details should be shown but no actions.
5. Test the print button; only the details should be shown and no other sidebar content.
6. Test adding the blocks to the sidebars for other post types. They should not be displayed on the frontend and the Widget editor should not crash on reload.
